gammapy/utils/nddata.py

```diff
@@ -0,0 +1,441 @@
+
+        Parameters
+        ----------
+        method : str {'linear', 'nearest'}
+            Interpolation method
+        kwargs : dict
+            Keys are the axis names, Values the evaluation points
+
+        Returns
+        -------
+        array : `~astropy.units.Quantity`
+            Interpolated values, axis order is the same as for the NDData array
+        """
+        for key in kwargs.keys():
+            if key not in self.axis_names:
+                raise ValueError('No axis for key {}'.format(key))
+
+        # Use nodes on unspecified axes
+        for name, val in zip(self.axis_names, self.axes):
+            kwargs.setdefault(name, val.nodes)
+
+        # Put in correct units
+        for key in kwargs:
+            val = Quantity(kwargs[key], unit=self.get_axis(key).unit)
+            kwargs[key] = np.atleast_1d(val).value
+
+        # Bring in correct order
+        values = [kwargs[_] for _ in self.axis_names]
+
+        # Apply log10 where necessary
+        # Todo : only calc log10 when needed
+        log_values = [np.log10(_) for _ in values]
+        loginterp = [a.log_interpolation for a in self.axes]
+        values = np.where(loginterp, log_values, values)
+
+        if method == 'linear':
+            return self._eval_linear(values, method='linear') * self.data.unit
+        elif method == 'nearest':
+            return self._eval_linear(values, method='nearest') * self.data.unit
+        else:
+            raise ValueError('Interpolator {} not available'.format(method))
+
+    def add_linear_interpolator(self, **kwargs):
+        """Add `~scipy.interpolate.RegularGridInterpolator`
+
+        The interpolation behaviour of an individual axis can be adjusted by
+        setting the ``log_interpolation`` attribute to True. After that the
+        interpolator has to be setup anew. TODO: Make this somehow more
+        convenient?
+
+        http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.interpolate.RegularGridInterpolator.html
+        """
+        kwargs.setdefault('bounds_error', False)
+
+        from scipy.interpolate import RegularGridInterpolator
+
+        points = [a._interp_nodes() for a in self.axes]
+        values = self.data.value
+
+        self._lininterp = RegularGridInterpolator(points, values, **kwargs)
+
+    def _eval_linear(self, values, method='linear'):
+        """Evaluate linear interpolator
+
+        Input: list of values to evaluate, in correct units and correct order.
+        """
+        if self._lininterp is None:
+            raise ValueError('Linear interpolation requested but no linear '
+                             'interpolator initialized')
+
+        shapes = [np.shape(_)[0] for _ in values]
+        points = list(itertools.product(*values))
+        res = self._lininterp(points, method=method)
+        res = np.reshape(res, shapes)
+
+        return res
+
+    def plot_image(self, ax=None, plot_kwargs = {}, **kwargs):
```

Comment:
Coverage looks pretty good:
https://coveralls.io/builds/6118260/source?filename=gammapy%2Futils%2Fnddata.py

This is the main method that's not tested.
I'd suggest adding a test that exercises this method (without assertions for now).